### PR TITLE
Update tests with using as clause

### LINKED ISSUE #1249: Make use of the ember-cli-page-object "as" DSL method
# Problem

From version 1.9.0, the `ember-cli-page-object` addon provides us with an `as` function to be used when testing individual elements in collections.

```javascript
andThen(() => {
  page.users(0).as(user => {
    assert.equal(user.name, 'John');
    assert.equal(user.lastName, 'Doe');
    assert.equal(user.email, 'john@doe');
  });
  
  page.users(1).as(user => {
    assert.equal(user.name, 'John');
    assert.equal(user.lastName, 'Doe');
    assert.equal(user.email, 'john@doe');
  });
  
  page.users(2).as(user => {
    assert.equal(user.name, 'John');
    assert.equal(user.lastName, 'Doe');
    assert.equal(user.email, 'john@doe');
  });
});
```

In several of our tests, we take an object from a page-object collection and store it into a variable to do these checks. We should make use of this helper instead.

Replace this:

```javascript
let someVar = page.collection(x);
assert.equal(someVar.someProperty, someValue);
//...
```

with this, where applicable:

```javascript
page.collection(x).as((someVar) => {
  assert.equal(someVar.someProperty, someValue);
});
```

### PR DESCRIPTION
**Pull request by npendery**
# What's in this PR?

## References
Fixes #1249

### CODE REVIEW

**Comment by JoshSmith** [comment]
Looks like some test failures. Rerunning now. Is this good to merge as-is, though, or are you still working?

**Comment by npendery** [comment]
@JoshSmith I think it's good to go, didn't see any other tests where this is useful

**Comment by JoshSmith** [comment]
@npendery cool! Maybe squash down then and I'll give it a final looksee?

File: tests/integration/components/stripe-connect-button-test.js [hunk 1]
```diff
@@ -13,6 +13,8 @@ test('it renders all required elements', function(assert) {
   this.set('url', url);
   this.render(hbs`{{stripe-connect-button url=url}}`);
 
-  assert.equal(this.$('.stripe-connect').attr('href'), url);
-  assert.equal(this.$('.stripe-connect').text(), 'Connect with Stripe');
+  this.$('.stripe-connect').as((component) => {
+    assert.equal(component.attr('href'), url);
+    assert.equal(component.text(), 'Connect with Stripe');
+  });
```
Comment (begedin) [inline tests/integration/components/stripe-connect-button-test.js]: This one is failing for me. You're gonna have to introduce a page object for it, or, which is perfectly fine as an option, don't switch to using `as`here and open an issue to introduce the page object.

Resolving this followed by a rebase and squash into a single commit is enough to get this merged, I think.

**Comment by JoshSmith** [comment]
Looks good!

**Pull request merged**